### PR TITLE
feat(pagination): add pagination component

### LINKED ISSUE #36: Build the style guide (Components: Pagination)
The design team has built the Style guide, which can be found [here](https://www.figma.com/file/wtojDaoY2PJED946ZMIvsX/OSCA-style-guide%2Fpattern).

This issue is a chunk of #4, ensure to check it first.

- Build the Pagination component similar to the design and document all classes used
- Build with great focus on all states and responsiveness.

### PR DESCRIPTION
**Pull request by ashinzekene**
Waiting for @perriefidelis on the interaction
Implemented Google's implementation

#### What does this PR do
Add the pagination component as in the [style guide](https://www.figma.com/file/wtojDaoY2PJED946ZMIvsX/OSCA-style-guide%2Fpattern)

#### Description of Task to be completed
Creation of the pagination component

#### How should this be manually tested
Run `npm run storybook`

#### What are the relevant issues this PR belongs to
Resolves #36 

#### Screenshots
image

image

### CODE REVIEW

File: src/components/pagination/pagination.stories.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react";
+import Pagination from ".";
+
+export default { title: "Pagination" };
```
Comment (BolajiAyodeji) [inline src/components/pagination/pagination.stories.js]: Please add the component name so the storybook docs addon can work effectively, see #41

```js
export default {
    title: 'Pagination',
    component: 'Pagination'
};
```

**Comment by perriefidelis** [comment]
Looks good, thanks @ashinzekene; @BolajiAyodeji 👍🏽

**Comment by ashinzekene** [comment]
@perriefidelis Check this. When there are more than 10 pages
1. The direction arrows
2. The pagination buttons showing

image

**Comment by perriefidelis** [comment]
looks good

**Comment by ashinzekene** [comment]
Cool @perriefidelis 

@BolajiAyodeji let's get #46 merged first. I've got some refactoring to do

**Comment by BolajiAyodeji** [comment]
Is this ready for a MR?

**Comment by ashinzekene** [comment]
Yeah, just fixed a merge conflict

**Pull request merged**